lib/contentful/resource_builder.rb

```diff
@@ -30,271 +28,119 @@ class ResourceBuilder
     # @see _ README for more information on Entry Mapping
     DEFAULT_ENTRY_MAPPING = {}
 
-    attr_reader :client, :response, :resource_mapping, :entry_mapping, :resource
+    attr_reader :json, :default_locale, :endpoint, :depth, :localized, :resource_mapping, :entry_mapping, :resource
 
-    def initialize(client,
-                   response,
-                   resource_mapping = {},
-                   entry_mapping = {},
-                   default_locale = Contentful::Client::DEFAULT_CONFIGURATION[:default_locale])
-      @response = response
-      @client = client
-      @included_resources = {}
-      @known_resources = Hash.new { |h, k| h[k] = {} }
-      @default_locale = default_locale
-      @resource_mapping = default_resource_mapping.merge(resource_mapping)
-      @entry_mapping = default_entry_mapping.merge(entry_mapping)
+    def initialize(json, configuration = {}, localized = false, depth = 0, endpoint = nil)
```

Comment:
The configuration object here is just to reduce the amount of things sent around, also, the same configuration options are required to be sent around to other places, so instead of having a thousand parameters, having a hash of configuration options seemed more reasonable.

Also, these options are internal, and non-required if someone wants to create a custom class